File: sc-frontend/src/features/question-bank/types/practiceSession.ts
export interface PracticeSession {
  id: string
  questionBankId: string
  courseId: string
  sysUserId: string
  sessionType: number
  totalQuestions: number
  answeredCount: number
  correctCount: number
  totalScore: number
  earnedScore: number
  startedAt: string
  completedAt: string | null
  status: number
  answers: PracticeAnswer[] | null
}

export interface PracticeAnswer {
  id: string
  sessionId: string
  questionId: string
  selectedOptionIds: string[] | null
  textAnswer: string | null
  isCorrect: number
  earnedScore: number
  answeredAt: string
}

export interface CreatePracticeSessionRequest {
  questionBankId: string
  courseId: string
  sessionType: number
}

export interface SubmitAnswerRequest {
  questionId: string
  selectedOptionIds?: string[]
  textAnswer?: string
}

export const SessionStatus: Record<number, string> = {
  0: '进行中',
  1: '已完成',
  2: '已放弃',
}

export const SessionType: Record<number, string> = {
  0: '顺序练习',
  1: '随机练习',
}

File: sc-course/src/main/java/com/dayz/sc/course/model/entity/PracticeSession.java
package com.dayz.sc.course.model.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

/**
 * PracticeSession 相关定义
 *
 * @author DaYZ
 * @since 2026-06-13
 */
@Getter
@Setter
@TableName("edu_practice_session")
public class PracticeSession {

    @TableId(type = IdType.INPUT)
    private UUID id;

    private UUID questionBankId;

    private UUID courseId;

    private UUID sysUserId;

    private Integer sessionType;

    private Integer totalQuestions;

    private Integer answeredCount;

    private Integer correctCount;

    private BigDecimal totalScore;

    private BigDecimal earnedScore;

    @TableField(fill = FieldFill.INSERT)
    private Instant startedAt;

    private Instant completedAt;

    private Integer status;
}
